File: Backend/app/src/main/java/com/example/demo/model/UserPlaylist.java
package com.example.demo.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.Instant;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "user_playlist")
public class UserPlaylist {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Long userId;
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    private String privacy; // PUBLIC, PRIVATE, UNLISTED

    private Long createdAt;

    @PrePersist
    public void onCreate() {
        this.createdAt = Instant.now().toEpochMilli();
    }
}

File: Mobile/app/src/main/java/com/example/myapplication/data/model/UserPlaylist.kt
package com.example.myapplication.data.model

data class UserPlaylist(
    val id: Long? = null,
    val userId: Long,
    val title: String? = null,
    val description: String? = null,
    val privacy: String? = null,   // PUBLIC, PRIVATE, UNLISTED
    val createdAt: Long? = null
)